lib/moss_theme/src/index.ts

```diff
@@ -1,109 +1,83 @@
-    "--color-windows-close-button-background": theme.colors.windowsCloseButtonBackground || "",
-    "--color-window-controls-linux-background": theme.colors.windowControlsLinuxBackground || "",
-    "--color-window-controls-linux-text": theme.colors.windowControlsLinuxText || "",
-    "--color-window-controls-linux-hover-background": theme.colors.windowControlsLinuxHoverBackground || "",
-    "--color-window-controls-linux-active-background": theme.colors.windowControlsLinuxActiveBackground || "",
-  };
+  const colorKeys = Object.keys(theme.colors) as (keyof Colors)[];
+
+  return colorKeys.reduce((acc, key) => {
+    const cssKey = `--color-${toKebabCase(String(key))}` as keyof ThemeCssVariables;
+    let value = theme.colors[key] || "";
+
+    if (value === "") {
+      value = findThemeColorValue(key, theme.colors);
+    }
+
+    acc[cssKey] = value;
+    return acc;
+  }, {} as ThemeCssVariables);
 }
 
 // Theme custom Tailwind color variables
-export const customTailwindColorVariables: Colors = {
-  primary: rgbaWithOpacity("--color-primary"),
-  sideBarBackground: rgbaWithOpacity("--color-sideBar-background"),
-  toolBarBackground: rgbaWithOpacity("--color-toolBar-background"),
-  pageBackground: rgbaWithOpacity("--color-page-background"),
-  statusBarBackground: rgbaWithOpacity("--color-statusBar-background"),
-  windowsCloseButtonBackground: rgbaWithOpacity("--color-windows-close-button-background"),
-  windowControlsLinuxBackground: rgbaWithOpacity("--color-window-controls-linux-background"),
-  windowControlsLinuxText: rgbaWithOpacity("--color-window-controls-linux-text"),
-  windowControlsLinuxHoverBackground: rgbaWithOpacity("--color-window-controls-linux-hover-background"),
-  windowControlsLinuxActiveBackground: rgbaWithOpacity("--color-window-controls-linux-active-background"),
-};
+export const customTailwindColorVariables: Record<keyof Colors, string> = Object.keys(new Colors()).reduce(
+  (acc, key) => {
+    const cssKey = `--color-${toKebabCase(key)}` as keyof ThemeCssVariables;
+    acc[key as keyof Colors] = rgbaWithOpacity(cssKey);
+    return acc;
+  },
+  {} as Record<keyof Colors, string>
+);
+
+type KebabCase<T extends string> = T extends `${infer F}${infer R}`
+  ? R extends Uncapitalize<R>
+    ? `${Lowercase<F>}${KebabCase<R>}`
+    : `${Lowercase<F>}-${KebabCase<R>}`
+  : T;
+
+function toKebabCase(str: string): string {
+  return str.replace(/[A-Z]/g, (letter) => `-${letter.toLowerCase()}`);
+}
+
+function findThemeColorValue(key: keyof Colors, keys: Colors): string {
+  const parts = String(key).split(/(?=[A-Z])/);
+  const len = parts.length;
+
+  for (let i = len - 1; i > 0; i--) {
+    parts[i] = parts[i].charAt(0).toLowerCase() + parts[i].slice(1);
+    const result = parts.slice(0, i).join("") + "." + parts.slice(i).join("");
+    if (keys[result] !== undefined) {
+      return keys[result];
+    }
+    parts[i] = parts[i].charAt(0).toUpperCase() + parts[i].slice(1);
+  }
+  return "";
+}
```

Comment:
we can try use map to process parts of the key, and instead of changing the case of characters and back, now w process each part of the key once. leave the first part unchanged, and convert the rest to camelCase

```typescript
function findThemeColorValue(key: keyof Colors, keys: Colors): string {
  const keyParts = String(key).split(/(?=[A-Z])/).map((part, index) => {
    return index === 0 ? part : part.charAt(0).toLowerCase() + part.slice(1);
  });

  const len = keyParts.length;

  for (let i = len - 1; i > 0; i--) {
    const camelCaseKey = keyParts.slice(0, i).join("") + "." + keyParts.slice(i).join("");

    if (keys[camelCaseKey] !== undefined) {
      return keys[camelCaseKey];
    }
  }

  return "";
}
```